features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt

```diff
@@ -52,10 +55,15 @@ class ConfigureRoomPresenter @Inject constructor(
     private val mediaPickerProvider: PickerProvider,
     private val mediaPreProcessor: MediaPreProcessor,
     private val analyticsService: AnalyticsService,
+    permissionsPresenterFactory: PermissionsPresenter.Factory,
 ) : Presenter<ConfigureRoomState> {
 
+    private val cameraPermissionPresenter: PermissionsPresenter = permissionsPresenterFactory.create(android.Manifest.permission.CAMERA)
+    private var pendingPermissionRequest = false
```

Comment:
Maybe this should be a `remember { mutableStateOf() }` ?

Moreover, perhaps this could also be hoisted to `PermissionsPresenter` to simplify the client side impementation.